index.js

```diff
@@ -36,7 +36,7 @@ module.exports = function(str, options) {
   }).join(newline);
 
   if (options.trim === true) {
-    result = result.replace(/[ \t]*$/gm, '');
+    result = result.replace(/\s+$/g, '');
```

Comment:
Not sure if you guys consider but, according to couple of checkers, this also a vulnerable regexp
https://devina.io/redos-checker
https://makenowjust-labs.github.io/recheck/playground/
<img width="659" alt="image" src="https://github.com/jonschlinkert/word-wrap/assets/5789670/c812647f-e659-45b0-878f-0faa0a1cd984">

Reply:
I think we can't do this with regex

Reply:
`String.trimEnd`

Reply:
I also think that the first commit was better.
But I think Node engines below v10 doesn't support stuff

Reply:
@aashutoshrathi 

Given we don't have a non-vulnerable regex solution and there's concern for older versions of Node.js, can we use this instead?

```
while(result.substring(result.length - 1) === ' ' || result.substring(result.length - 1) == '\t') {
    result = result.substring(0, result.length - 1)
}
```

Reply:
@aashutoshrathi @jonschlinkert @seahindeniz 

How about this?

`
result = result.replace(/([^ \t])([ \t]*)$/gm, '$1');
`

This assumes every line will start with something other than space or tab. The `$1` will preserve whatever that character is. I confirmed that this is not a vulnerable regex and that all unit tests pass.

Reply:
@aashutoshrathi have you tried this suggestion?

Reply:
not yet, will do

Reply:
This works just as well as .trimEnd();

```
$ time node -e 'payloadstr=""; for ( i=0; i < 100000; i++ ) payloadstr+="\t"; payloadstr+="a"; payloadstr.replace(/([^ \t])([ \t]*)$/gm, "$1");'

real    0m0.087s

```